File: app/src/main/java/com/example/inklink/models/Article.java
package com.example.inklink.models;

public class Article {
    private int id, userId, reportCount;
    private String articleTitle, articleContent, articleStatus, creationDate;

    public Article() {}

    public Article(String title, String content, int userId) {
        this.articleTitle = title;
        this.articleContent = content;
        this.userId = userId;
    }

    public int getId() { return id; }

    public void setId(int id) { this.id = id; }

    public int getUserId() { return userId; }

    public void setUserId(int userId) { this.userId = userId; }

    public int getReportCount() { return reportCount; }

    public void setReportCount(int reportCount) { this.reportCount = reportCount; }

    public String getArticleTitle() { return articleTitle; }

    public void setArticleTitle(String articleTitle) { this.articleTitle = articleTitle; }

    public String getArticleContent() { return articleContent; }

    public void setArticleContent(String articleContent) { this.articleContent = articleContent; }

    public String getArticleStatus() { return articleStatus; }

    public void setArticleStatus(String articleStatus) { this.articleStatus = articleStatus; }

    public String getCreationDate() { return creationDate; }

    public void setCreationDate(String creationDate) { this.creationDate = creationDate; }
}

File: app/src/main/java/com/example/inklink/models/Article.kt
package com.example.inklink.models

class Article {
    var id: Int = 0
    var userId: Int = 0
    var reportCount: Int = 0
    var articleTitle: String? = null
    var articleContent: String? = null
    var articleStatus: String? = null
    var creationDate: String? = null

    constructor()

    constructor(title: String, content: String, userId: Int) {
        this.articleTitle = title
        this.articleContent = content
        this.userId = userId
    }

    override fun toString(): String {
        return """
            Article: {
                id: ${this.id},
                userId: ${this.userId},
                reportCount: ${this.reportCount},
                articleTitle: ${this.articleTitle},
                articleContent: ${this.articleContent},
                articleStatus: ${this.articleStatus},
                creationDate: ${this.creationDate},
            }
        """.trimIndent()
    }
}
